src/facade/disk_backed_queue.h

```diff
@@ -0,0 +1,56 @@
+// Copyright 2025, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <io/file.h>
+
+#include <functional>
+#include <memory>
+#include <string_view>
+#include <system_error>
+
+namespace facade {
+
+class DiskBackedQueue {
+ public:
+  explicit DiskBackedQueue(uint32_t conn_id);
+  ~DiskBackedQueue();
+
+  std::error_code Init();
+
+  // Check if we can offload bytes to backing file.
+  bool HasEnoughBackingSpaceFor(size_t bytes) const;
+
+  std::error_code Push(std::string_view blob);
+
+  std::error_code Pop(std::string* out);
```

Comment:
it's not a good interface now - as you do not pop an element. You will usually have a destination buffer which you will want to fill till the end of its capacity or less. Similarly to socket Recv() interface.